### PR TITLE
[LAN-7196] use file client

### PR DESCRIPTION
**Pull request by jckhang**
Use file client directly without going through mmcv.FileClient

### CODE REVIEW

**Review by AsiaCao — APPROVED** [review]

File: mmdet/datasets/avi_custom.py [hunk 1]
```diff
@@ -426,124 +434,126 @@ def __repr__(self):
...
+
+        # TODO: hack to cache the xml file
+        # xml_dict = avi_io.read_dict_like(xml_path)
+        xml_dict = self.read_s3_with_cache(xml_path)
+        bboxes = []
+        labels = []
+        if "annotation" in xml_dict and xml_dict["annotation"] is not None:
+            meta = xml_dict["annotation"]
+        else:
+            meta = {}
```
Comment (Canas) [inline mmdet/datasets/avi_custom.py:485]: you can just do `meta = xml_dict.get('annotation', {})`

File: mmdet/datasets/avi_custom.py [hunk 2]
```diff
@@ -426,124 +434,126 @@ def __repr__(self):
...
-    label_id_map: Dict[str, int],
-    xml_path: str,
-    img_path: str,
-):
-    if xml_path == "":
-        img_pillow = Image.open(img_path)
+        width = int(round(float(meta["size"]["width"])))
+        height = int(round(float(meta["size"]["height"])))
+
+        if "object" in meta:
```
Comment (Canas) [inline mmdet/datasets/avi_custom.py:499]: we can probably move this to a separate method, like `parse_object` or `get_bboxes_and_labels` so we test apart

File: mmdet/datasets/avi_custom.py [hunk 3]
```diff
@@ -426,124 +434,126 @@ def __repr__(self):
...
+        paths = list(zip(media_ids, xml_paths, image_paths))
+        annotations = []
+        for media_id, xml_path, image_path in mmcv.track_iter_progress(paths):
+            annotations.append(
+                self.parse_xml(media_id, label_id_map, xml_path, image_path)
+            )
+
+        return annotations
+
+    def parse_xml(
```
Comment (Canas) [inline mmdet/datasets/avi_custom.py:453]: can we add a docstring explaining the expected content/keys for the XML?

File: mmdet/datasets/avi_custom.py [hunk 4]
```diff
@@ -140,6 +143,11 @@ def __init__(
             self._set_group_flag()
 
         # processing pipeline
+        for pipeline_ix in range(len(pipeline)):
```
Comment (Canas) [inline mmdet/datasets/avi_custom.py:146]: nit: you can actually do `for idx, pipeline in enumerate(pipeline)` to improve readability

**Pull request merged**